src/Data/Vec/Instances.agda

```diff
@@ -9,7 +9,16 @@
 module Data.Vec.Instances where
 
 open import Data.Vec.Categorical
+open import Data.Vec.Properties
+  using (≡-dec)
+open import Relation.Binary.PropositionalEquality.Properties
+  using (isDecEquivalence)
+open import Data.Vec.Relation.Binary.Equality.DecPropositional
+open import Relation.Binary.TypeClasses
+
 
 instance
   vecFunctor = functor
   vecApplicative = applicative
+
+  ≡-isDecEquivalence-Vec = λ {a} {A} {{≡-isDecEquivalence-A}} {n} → isDecEquivalence (≡-dec {a} {A} (_≟_ {{≡-isDecEquivalence-A}}) {n})
```

Comment:
Out curiosity: here and in a few other places, you are very explicit about implicits - why?

Reply:
It's mainly laziness; the alternative is writing down the actual type signature of `≡-isDecEquivalence-Vec` (which might actually be a better idea now that I think about it..)

Reply:
Yup, I'd vote for having the type signature instead. Much more readable!

Reply:
I just added the type signatures and it really does make a big difference. Not sure *what* I was thinking yesterday!

Reply:
Thanks for reinforcing those instincts of mine that underlie asking the question. :smirk: